### PR TITLE
AP_Arming: remove unused arming_method member

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AP_Arming/AP_Arming.cpp [hunk 1]
```diff
@@ -83,8 +83,7 @@ AP_Arming::AP_Arming(const AP_AHRS &ahrs_ref, Compass &compass,
     ahrs(ahrs_ref),
     _compass(compass),
     _battery(battery),
-    armed(false),
-    arming_method(NONE)
+    armed(false)
```
Comment (WickedShell) [inline libraries/AP_Arming/AP_Arming.cpp]: Unneeded initializer

Comment (peterbarker) [reply libraries/AP_Arming/AP_Arming.cpp]: Yes, I saw this pointless initialiser.  I thought, "I'll remove that when I change this to use singletons".

Comment (WickedShell) [reply libraries/AP_Arming/AP_Arming.cpp]: My thought was: you are already modifying the line, no reason to not just delete it :)

**Comment by peterbarker** [comment]
@amilcarlucas It didn't use an enum, this the hilarious conversion between `bool` and `uint8_t` in the call structure - still present!

I do plan on creating a PR to use the existing enumeration, however - passing the arming method around is critical, we should just add some type safety.

**Comment by peterbarker** [comment]
I'm having second thoughts about this removal given the presence of Copter's `ap.armed_with_switch`.

OTOH - I'm not sure how currently/should behave if someone arms with the rudder, flicks the arming switch and then sends an mavlink arming command.

**Comment by peterbarker** [comment]
On Sun, 24 Jun 2018, WickedShell wrote:

> My thought was: you are already modifying the line, no reason to not just delete it :)

Entirely reasonable.  But when you start pulling on threads... :-)

I've PR'd the removal as part of a separate PR now.

**Pull request closed without merging**